framework/helpers/BaseArrayHelper.php

```diff
@@ -203,17 +204,28 @@ public static function getValue($array, $key, $default = null)
             $key = substr($key, $pos + 1);
         }
 
-        if (is_object($array)) {
+        if (static::isArrayAccess($array)) {
+            return static::arrayHasKey($array, $key) ? $array[$key] : $default;
+        } elseif (is_object($array)) {
             // this is expected to fail if the property does not exist, or __get() is not implemented
             // it is not reliably possible to check whether a property is accessible beforehand
             return $array->$key;
-        } elseif (is_array($array)) {
-            return (isset($array[$key]) || array_key_exists($key, $array)) ? $array[$key] : $default;
         }
 
         return $default;
     }
 
+    private static function arrayHasKey($array, $key)
```

Comment:
Does it worth making it public?

Reply:
yes, I can actually make it public if you want

Reply:
Well... if you find it useful.

Reply:
I think I will much rather fix `BaseArrayHelper::keyExists()` method that this class already has...